iped-carvers/iped-carvers-impl/src/main/java/iped/carvers/custom/TorrentCarver.java

```diff
@@ -0,0 +1,62 @@
+package iped.carvers.custom;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+
+import com.dampcake.bencode.BencodeInputStream;
+
+import iped.carvers.api.Hit;
+import iped.carvers.standard.DefaultCarver;
+import iped.data.IItem;
+import iped.io.SeekableInputStream;
+import iped.utils.IOUtil;
+
+public class TorrentCarver extends DefaultCarver {
+    @Override
+    public boolean isSpecificIgnoreCorrupted() {
+        return true;
+    }
+
+    @Override
+    public long getLengthFromHit(IItem parentEvidence, Hit header) throws IOException {
+        SeekableInputStream is = null;
+        BencodeInputStream bis = null;
+        try {
+            long off = header.getOffset();
+            is = parentEvidence.getSeekableInputStream();
+            is.seek(header.getOffset());
+            bis = new BencodeInputStream(is, StandardCharsets.UTF_8, true);
+            Map<String, Object> dict = bis.readDictionary();
+            if (isValid(dict)) {
+                long len = is.position() - off;
+                if (len <= 0) {
+                    len = -1;
+                } else {
+                    Long parentLen = parentEvidence.getLength();
+                    if (parentLen != null && off + len > parentLen) {
+                        len = parentLen - off;
```

Comment:
You are right, it should never reach that condition.
It was a protection to avoid going beyond the parent file end, as `Item.getSeekableInputStream()` could allow that, in a very specific situation.
You can remove it, if you prefer.

Reply:
It's rare, but it's possible that the returned InputStream is smaller or larger than the item length, I have seen it in the past when getting the InputStream from sleuthkit, e.g., when FS is corrupted.

Reply:
Right. No problem with the protection code, it can stay where it is. I was just being sure if I was missing something, as I am preparing test sample to reach the most possible code execution.